packages/lesswrong/components/users/EditProfileForm.tsx

```diff
@@ -30,25 +30,39 @@ const EditProfileForm = ({classes}: {
 }) => {
   const currentUser = useCurrentUser()
   const { history } = useNavigation()
+  const { params } = useLocation()
   
-  const { Typography, WrappedSmartForm } = Components;
+  const { Typography, WrappedSmartForm } = Components
   
-  if (!currentUser) {
+  const terms: {slug?: string, documentId?: string} = params.slug ?
+    { slug: params.slug } :
+    currentUser ? { documentId: currentUser._id } : {}
```

Comment:
I would avoid this nested ternary
```suggestion
  let terms: {slug?: string, documentId?: string} = {}
  if (params.slug) {
    terms.slug = params.slug
  } else if (currentUser) {
    terms.documentId = currentUser._id
  }
```